monai/handlers/clearml_handlers.py

```diff
@@ -0,0 +1,175 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Mapping, Sequence
+
+from monai.utils import optional_import
+
+from .tensorboard_handlers import TensorBoardImageHandler, TensorBoardStatsHandler
+
+if TYPE_CHECKING:
+    from clearml import Task
+else:
+    Task, _ = optional_import("clearml", name="Task")
+
+
+class ClearMLHandler:
+    """
+    Base class for the handlers to log everything to ClearML.
+    For more details of ClearML usage, please refer to:
+    https://clear.ml/docs/latest/docs/references/sdk/task
+
+    Usage example is available in the tutorial:
+    https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/unet_segmentation_3d_ignite.ipynb.
+
+    """
+
+    def __init__(
+        self,
+        project_name: str | None,
+        task_name: str | None,
+        output_uri: str | bool,
+        tags: Sequence[str] | None,
+        reuse_last_task_id: bool,
+        continue_last_task: bool,
+        auto_connect_frameworks: bool | Mapping[str, bool | str | list],
+        auto_connect_arg_parser: bool | Mapping[str, bool],
+    ) -> None:
+        """
+        Args:
+            project_name: ClearML project name, default to 'MONAI'.
+            task_name: ClearML task name, default to 'monai_experiment'.
+            output_uri: The default location for output models and other artifacts, default to 'True'.
+            tags: A list of tags (str) to the created Task, default to 'None'.
+            reuse_last_task_id: Force a new Task (experiment) with a previously used Task ID, default to 'True'.
+            continue_last_task: Continue the execution of a previously executed Task (experiment), default to 'False'.
+            auto_connect_frameworks: Automatically connect frameworks, default to 'True'.
+            auto_connect_arg_parser: Automatically connect an argparse object to the Task, default to 'True'.
+
+        """
+
+        if Task.current_task():
```

Comment:
We don't really need to import clearml before this line, could you please move the optional import of `Task` here?

Otherwise the clearml patching logic for all optional packages is applied every time the user imports monai, I don't think we should introduce this behaviour before it's well tested in monai https://github.com/allegroai/clearml/blob/37dde23bc94d297d6731392db61b9a5f33eaece3/clearml/binding/fire_bind.py#L328

Reply:
@wyli , Fixed optional_import location according to your suggestion.

Reply:
Thank you, please revise the minor style issue, then  I'll trigger the tests and merge...

Reply:
@wyli , Type hinting fixed.